app/screens/channel_info/channel_info.js

```diff
@@ -90,27 +91,37 @@ export default class ChannelInfo extends PureComponent {
         this.props.actions.getCustomEmojisInText(this.props.currentChannel.header);
     }
 
-    componentWillReceiveProps(nextProps) {
-        if (this.props.theme !== nextProps.theme) {
-            setNavigatorStyles(this.props.componentId, nextProps.theme);
-        }
-
-        let isFavorite = this.state.isFavorite;
+    static getDerivedStateFromProps(nextProps, state) {
+        let isFavorite = state.isFavorite;
         if (isFavorite !== nextProps.isFavorite) {
             isFavorite = nextProps.isFavorite;
         }
 
-        let isMuted = this.state.isMuted;
+        let isMuted = state.isMuted;
         if (isMuted !== nextProps.isChannelMuted) {
             isMuted = nextProps.isChannelMuted;
         }
 
-        let ignoreChannelMentions = this.state.ignoreChannelMentions;
+        let ignoreChannelMentions = state.ignoreChannelMentions;
         if (ignoreChannelMentions !== nextProps.ignoreChannelMentions) {
             ignoreChannelMentions = nextProps.ignoreChannelMentions;
         }
 
-        this.setState({isFavorite, isMuted, ignoreChannelMentions});
+        if (nextProps.currentUserId !== state.prevPropsUserId) {
```

Comment:
not following, why returning the state only when the currentUserId changes? this won't even happen as the currentUserId does not changes while you are logged in with the same user.

Reply:
Should I just simply return the state here? But then, state updates would be lost (I think so).

Reply:
Perhaps we could just remove `getDerivedStateFromProps` entirely? We already set the state from the props when this screen is opened, and the click handlers for each of those settings update the state when the user taps on them, so the only case we'd really lose out on is if the user updates one of those from another device while this screen is open which is virtually never going to happen